bin/osg-topology

```diff
@@ -166,48 +193,140 @@ def list_security_contacts(args):
-                            if contact_list_type == 'Security Contact':
-                                security_contacts = contact_list_info
-
-                if name and security_contacts and (not args.name_filter or fnmatch.fnmatch(name, args.name_filter) or args.name_filter in name):
-                    keys = results.setdefault(name, set())
-                    for contact in security_contacts:
-                        key = None
-                        if 'Name' in contact and 'Email' in contact:
-                            key = "%s<%s>" % (contact['Name'], contact['Email'])
-                        elif 'Name' in contact:
-                            key = contact['Name']
-                        if key:
-                            keys.add(key)
-
-    sites = results.keys()
-    sites.sort()
-    for site in sites:
-        keys = list(results[site])
-        keys.sort()
-        print("- Resource: %s" % site)
-        for key in keys:
-             print(u"  * %s" % key)
+                            if contact_list.tag == 'ContactList':
+                                contact_list_info.extend( \
+                                    get_contact_list_info(contact_list))
+
+                if name and contact_list_info:
+                    results[name] = contact_list_info
+
+    return results
+
+
+def list_resource_contacts(args):
+    """
+    List resource contacts for OSG.
+    """
+    results = get_resource_contacts(args)
+    print_contacts(args, 'Resource', results)
+
+
+def get_vo_contacts(args):
+    """
+    Get resource contacts for OSG.  Return results.
+    """
+    root = get_contacts(args, 'vo', 'VO')
+    if root == None:
+        return 1
+
+    results = {}
+    for child_vo in root:
+        if child_vo.tag != "VO":
+            print("MyOSG returned a non-VO (%s) inside summary." % \
+                  root.tag, file=sys.stderr)
+            return 1
+        name = None
+        contact_list_info = []
+        for item in child_vo:
+            if item.tag == 'Name':
+                name = item.text
+            if item.tag == "ContactTypes":
+                for contact_type in item:
+                    contact_list_info.extend( \
+                        get_contact_list_info(contact_type))
+
+        if name and contact_list_info:
+            results[name] = contact_list_info
+
+    return results
+
+
+def list_vo_contacts(args):
+    """
+    List VO contacts for OSG.
+    """
+    results = get_vo_contacts(args)
+    print_contacts(args, 'VO', results)
+
+
+def list_contacts(args):
+    """
+    List resource and VO contacts for OSG.
+    """
+    if args.output_mode == "email":
+        results = get_resource_contacts(args)
+        results.update(get_vo_contacts(args))
+        print_contacts(args, 'combined', results)
+    else:
+        list_resource_contacts(args)
+        list_vo_contacts(args)
+
+
+def print_contacts(args, type, results):
```

Comment:
Can we rename `type` so we don't override the built-in?

Reply:
Done